coaster/views/classview.py

```diff
@@ -586,21 +594,14 @@ def loader(self, kwargs):
                                 else:
                                     attr = attr.property.argument
                                 if attr not in joined_models:
-                                    # SQL JOIN the other model
-                                    query = query.join(attr)
+                                    # SQL JOIN the other model on the basis of
+                                    # the relationship that led us to this join
+                                    query = query.join(attr, relattr)
```

Comment:
How does this work as `attr` and `relattr` are the same?

Reply:
`attr` is replaced when following a dotted path. `relattr` is not.

Reply:
This distinction is crucial when a model has two foreign keys to another model. We need to supply the specific attribute to join on.